File: bridge/python/tests/lint.py
from destack import LintRequest, Scope

from session import open_workspace


def test_lint_accepts_module() -> None:
    fixture = open_workspace(
        [
            ("destack.json", '{"name":"@test/app"}'),
            ("src/index.ds", "export const value = 1;"),
        ],
    )
    workspace = fixture.workspace
    revision = workspace.revision()
    module = workspace.module("src/index.ds")
    profile = workspace.profile(revision, module, "js")

    # lint the loaded module profile
    output = workspace.lint(revision, LintRequest(Scope.module(module, profile)))

    assert output.diagnostics == []

File: bridge/rust/tests/lint.rs
#[path = "support/session.rs"]
mod session;

use session::open_workspace;

#[test]
fn test_lint_accepts_module() -> destack::Result<()> {
    let fixture = open_workspace(&[
        ("destack.json", r#"{"name":"@test/app"}"#),
        ("src/index.ds", "export const value = 1;"),
    ])?;
    let workspace = fixture.workspace;
    let revision = workspace.revision()?;
    let module = workspace.module("src/index.ds")?;
    let profile = workspace.profile(revision, module, "js")?;

    // lint the loaded module profile
    let output = workspace.lint(
        revision,
        destack::LintRequest::new(destack::Scope::module(module, profile)),
    )?;

    assert_eq!(output.diagnostics, []);

    Ok(())
}
